### PR TITLE
tests: drivers: can: move CAN utilities tests to correct location and add DLC utility tests

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Move the CAN utilities tests to the tests/drivers/can/ directory. The tests/subsys/canbus directory is for tests related to code located in subsys/canbus. Add tests for can_dlc_to_bytes() and can_bytes_to_dlc() utility functions.

### CODE REVIEW

**Review by alexanderwachter — APPROVED** [review]

File: tests/drivers/can/utilities/src/main.c [hunk 1]
```diff
@@ -135,13 +149,58 @@ static void test_zcan_filter_to_can_filter(void)
...
+
+	/* CAN-FD DLC, 12 to 64 data bytes in steps */
+	zassert_equal(can_dlc_to_bytes(9),  12, "wrong number of bytes for DLC 9");
+	zassert_equal(can_dlc_to_bytes(10), 16, "wrong number of bytes for DLC 10");
+	zassert_equal(can_dlc_to_bytes(11), 20, "wrong number of bytes for DLC 11");
+	zassert_equal(can_dlc_to_bytes(12), 24, "wrong number of bytes for DLC 12");
+	zassert_equal(can_dlc_to_bytes(13), 32, "wrong number of bytes for DLC 13");
+	zassert_equal(can_dlc_to_bytes(14), 48, "wrong number of bytes for DLC 14");
+	zassert_equal(can_dlc_to_bytes(15), 64, "wrong number of bytes for DLC 15");
+}
```
Comment (str4t0m) [inline tests/drivers/can/utilities/src/main.c:172]: You could consider additionally adding one test for an out of range dlc like 16 or higher.

**Review by str4t0m — APPROVED** [review]

**Review by karstenkoenig — APPROVED** [review]
